internal/brokers/withexamples.go

```diff
@@ -0,0 +1,116 @@
+//go:build withexamplebroker
+
+package brokers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/ubuntu/authd/internal/brokers/examplebroker"
+	"github.com/ubuntu/authd/internal/log"
+)
+
+// useExampleBrokers starts a mock system bus and exports the examplebroker in it.
+func useExampleBrokers() (string, error) {
+	busCleanup, err := startSystemBusMock()
+	if err != nil {
+		return "", err
+	}
+	log.Debugf(context.Background(), "Mock system bus started on %s\n", os.Getenv("DBUS_SYSTEM_BUS_ADDRESS"))
+
+	// Create the directory for the broker configuration files.
+	cfgPath, err := os.MkdirTemp(os.TempDir(), "examplebroker.d")
+	if err != nil {
+		busCleanup()
+		return "", err
+	}
+	cleanup := func() {
+		os.RemoveAll(cfgPath)
+		busCleanup()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		if err = examplebroker.StartBus(ctx, cfgPath); err != nil {
+			fmt.Printf("Error starting example broker: %v\n", err)
+			cleanup()
+		}
+	}()
+	// Give some time for the broker to start
+	time.Sleep(time.Second)
+
+	go func() {
+		time.Sleep(time.Minute)
+		cancel()
+		<-done
+		cleanup()
+		fmt.Println("Stopped example broker")
+	}()
+
+	return cfgPath, nil
+}
+
+// startSystemBusMock starts a mock dbus daemon and returns a cancel function to stop it.
+func startSystemBusMock() (func(), error) {
+	busDir := filepath.Join(os.TempDir(), "authd-system-bus-mock")
+	if err := os.MkdirAll(busDir, 0750); err != nil {
+		return nil, err
+	}
+
+	cfgPath := filepath.Join(busDir, "bus.conf")
+	listenPath := filepath.Join(busDir, "bus.sock")
+
+	err := os.WriteFile(cfgPath, []byte(fmt.Sprintf(systemBusMockCfg, listenPath)), 0600)
+	if err != nil {
+		err = errors.Join(err, os.RemoveAll(busDir))
+		return nil, err
+	}
+
+	busCtx, busCancel := context.WithCancel(context.Background())
+	//#nosec:G204 // This is a test helper and we are in control of the arguments.
```

Comment:
change the comment maybe to tell this is only for demo purpose and don’t end up in production code?